config/number.c

```diff
@@ -49,16 +49,19 @@
  */
 static intptr_t native_get(void *var)
 {
-  return (*(intptr_t *) var & TOGGLE_BIT) ? 0 : *(short *) var;
+  // take care of endianess and always read intptr_t value
+  intptr_t v = *(intptr_t *) var;
+  return (v & TOGGLE_BIT) ? 0 : (short) v;
```

Comment:
@flatcap we were debugging this yesterday (I only saw you had already merged the fix today) and the `(short) v` didn't click for us. Shouldn't it return `intptr_t` and the conversion to `short` be done explicitly by the caller?

Reply:
Ok, just saw `#define TOGGLE_BIT ((SHRT_MAX + 1) << 1)`, sorry